### PR TITLE
Build node images with standardised format

### LINKED ISSUE #1084: Standardise naming of nodejs images
To not fill the PR with non-related discussion, re: https://github.com/GoogleContainerTools/distroless/pull/1083#issuecomment-1184551777, I open this issue as a tracker to discuss standardising nodejs images.

> I think it might be time to standardize the naming of nodejs images to be similar to the others.
> ```
> nodejs<version>-debian<version>:{root, nonroot, debug, debug-nonroot}
> ```
> We can leave the old naming as legacy for now and deprecate those in ~6 months ?

### PR DESCRIPTION
**Pull request by omBratteng**
Standardises the nodejs images.

Closes #1084

### CODE REVIEW

**Comment by omBratteng** [comment]
Just a first draft @loosebazooka, not that happy with having two blocks of `NODEJS`, one with distro suffix, and one without.
Any clever ideas how to merge them into one?

As far as I understand, the `container_push` rules pushes the images to the image registry like `nodejs<version>-debian<version>:latest-<arch>`, and images have to be pushed to registry to be able to create manifests.
Perhaps in the manifest creation script, we can use the `nodejs<version>-debian<version>:latest-<arch>` images to create the `nodejs<version>:latest` manifests, without having to have its own `NODEJS` block?

File: BUILD [hunk 1]
```diff
@@ -125,11 +125,40 @@ PYTHON3.update({
...
+        ("latest", ""),
+        ("debug", "_debug"),
+    ]
+})
+
+NODEJS.update({
     "{REGISTRY}/{PROJECT_ID}/nodejs:14": "//nodejs:nodejs14_amd64_debian11",
     "{REGISTRY}/{PROJECT_ID}/nodejs:14-debug": "//nodejs:nodejs14_debug_amd64_debian11",
     "{REGISTRY}/{PROJECT_ID}/nodejs-debian11:14": "//nodejs:nodejs14_amd64_debian11",
     "{REGISTRY}/{PROJECT_ID}/nodejs-debian11:14-debug": "//nodejs:nodejs14_debug_amd64_debian11",
```
Comment (omBratteng) [inline BUILD:160]: Don't see a point in converting this to support multi-arch for the old image naming scheme, just for a couple of months. So I think I'll just leave this be. But adding it for the new naming scheme.

**Comment by loosebazooka** [comment]
Is this ready for review?

**Comment by omBratteng** [comment]
I don't remember, had some vacation. I think there still were some changes I wanted to do with the cloudbuild docker script, but I think that would affect more images, and is suitable for its own PR.

**Comment by emtunc** [comment]
How's this looking, team?

**Comment by omBratteng** [comment]
I will add the nonroot versions in a separate PR

**Comment by omBratteng** [comment]
The examples are failing, as they're trying to pull the images this PR will publish

**Comment by emtunc** [comment]
Do the examples script(s) need to be modified to help this pass or can it be merged? I see https://github.com/GoogleContainerTools/distroless/pull/1083 is also dependent on this PR

**Comment by omBratteng** [comment]
The test is failing, because it is trying to use the new images made with this PR.
If the goal is all checks passed, I could undo the example tests, and then open a new PR for those after this is merged.

I will rebase #1083 once this is merged, as it is dependent on the changes done here.

**Comment by loosebazooka** [comment]
I think that's fine, we can let that test fail.

**Review by loosebazooka — APPROVED** [review]
lgtm... I'll send out an announce and we can let users know to switch over to new node tags. (tags wont deprecate for 6 months)

**Comment by loosebazooka** [comment]
> I think that's fine, we can let that test fail.

just kidding, yeah lets move that test update to a new PR.

**Comment by omBratteng** [comment]
I have reverted the tests to use the old tags, and I'll prepare a PR to update the tests.

**Comment by omBratteng** [comment]
> (tags wont deprecate for 6 months)

Which aligns up closely with the EOL of node14

> How about deprecate the old naming 2023-04-30, that matches with the EOL of node 14, so we can drop the non-manifested builds too.
> https://github.com/GoogleContainerTools/distroless/pull/1083#issuecomment-1184555454

**Comment by loosebazooka** [comment]
kk lets see how this goes.

**Pull request merged**